src/main/java/net/minecraftforge/common/SimpleAnvilRecipe.java

```diff
@@ -0,0 +1,136 @@
+package net.minecraftforge.common;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.minecraft.block.Block;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.world.World;
+import net.minecraftforge.oredict.OreDictionary;
+
+public class SimpleAnvilRecipe implements IAnvilRecipe
+{
+    protected ArrayList<Object> inputs = new ArrayList<Object>();
+    protected final ItemStack   output;
+    protected final int         expCost;
+    protected final int         matCost;
+
+    public SimpleAnvilRecipe(ItemStack result, Object inputLeft, Object inputRight, int materialCost, int experienceCost)
+    {
+        this.output = result.copy();
+        this.matCost = materialCost;
+        this.expCost = experienceCost;
+        Object left = processInput(inputLeft);
+        Object right = processInput(inputRight);
+        if (left == null || (right == null && inputRight != null))
+        {
+            String ret = "Invalid simple anvil recipe: " + inputLeft + ", " + inputRight + ", " + this.output;
+            throw new RuntimeException(ret);
+        }
+        this.inputs.add(left);
+        this.inputs.add(right);
+    }
+
+    private Object processInput(Object input)
+    {
+        if (input instanceof String)
+        {
+            return OreDictionary.getOres((String)input);
+        }
+        else if (input instanceof ItemStack)
+        {
+            return ((ItemStack)input).copy();
+        }
+        else if (input instanceof Item)
+        {
+            return new ItemStack((Item)input, 1, OreDictionary.WILDCARD_VALUE);
+        }
+        else if (input instanceof Block)
+        {
+            return new ItemStack((Block)input, OreDictionary.WILDCARD_VALUE);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean matches(ItemStack inputLeft, ItemStack inputRight, String newName, World world)
+    {
+        return checkMatch(this.inputs.get(0), inputLeft, false) && checkMatch(this.inputs.get(1), inputRight, true);
+    }
+
+    private boolean checkMatch(Object target, ItemStack tableItem, boolean checksize)
```

Comment:
IMO `Object target` can be changed to `List<ItemStack> target` and make life easier if you accept my suggestion.